### PR TITLE
Converts DbtCloudCredentials to a block

### PR DESCRIPTION
**Pull request by desertaxle**
## Summary

- Converts `DbtCloudCredentials` from a `dataclass` to a `Block` to take allow the storage of dbt Cloud credentials via Prefect.

## Checklist
- [ ] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-dbt/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_dbt/cloud/credentials.py [hunk 1]
```diff
@@ -24,9 +24,36 @@ class DbtCloudCredentials:
...
+
+
+        trigger_job_run_flow()
+        ```
     """
 
-    api_key: str
+    _block_type_name = "dbt Cloud Credentials"
+    _logo_url = "https://asset.brandfetch.io/idofJOT4bu/idxrwTdDC-.svg"
+    _code_example = """/
```
Comment (ahuang11) [inline prefect_dbt/cloud/credentials.py:49]: Maybe use of `from textwrap import dedent`?

File: setup.py [hunk 2]
```diff
@@ -27,6 +27,9 @@
     python_requires=">=3.7",
     install_requires=install_requires,
     extras_require={"dev": dev_requires},
+    entry_points={
+        "prefect.collections": ["DbtCloudCredentials = prefect_dbt.cloud.credentials"]
```
Comment (ahuang11) [inline setup.py:31]: Fancy

File: prefect_dbt/cloud/credentials.py (hunk 1 above)
Comment (desertaxle) [reply prefect_dbt/cloud/credentials.py:49]: Yep, the `Block` base class handles that for us: https://github.com/PrefectHQ/orion/blob/main/src/prefect/blocks/core.py#L301-L303

File: setup.py (hunk 2 above)
Comment (desertaxle) [reply setup.py:31]: This is the piece that allows the `DbtCloudCredentials` block to be discovered by the core library and shown in the UI.

**Review by ahuang11 — APPROVED** [review]

**Pull request merged**